trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/testgen/Req2CauseTrackingPea.java

```diff
@@ -194,18 +198,18 @@ private static boolean isEffectTransition(final Phase source, final Transition t
 				&& transition.getGuard().and(effectCdd.negate().prime(Collections.emptySet())) == CDD.FALSE;
 	}
 
-	private static void setFlags(final Phase[] initialPhases) {
+	private static void setFlags(final List<Phase> initialPhases) {
 		for (final Phase p : initialPhases) {
 			p.isInit = true;
 		}
 	}
 
-	private Phase[] transformLocations(final PhaseEventAutomata oldPea, final IReqSymbolTable reqSymbolTable,
+	private List<Phase> transformLocations(final PhaseEventAutomata oldPea, final IReqSymbolTable reqSymbolTable,
 			final Set<String> effectVars, final int effectPhase, final DCPhase[] dcFormula) {
-		final Phase[] oldPhases = oldPea.getPhases();
-		final Phase[] newPhases = new Phase[2 * oldPhases.length];
-		for (int i = 0; i < oldPhases.length; i++) {
-			final Phase oldPhase = oldPhases[i];
+		final List<Phase> oldPhases = oldPea.getPhases();
+		final List<Phase> newPhases = new ArrayList<>();
```

Comment:
You should specify the initial capacity of an ArrayList, like in the old code.

Reply:
It might make more sense to let newPhases be an array, and then  return Arrays.toList(newPhases).